text/1004-transactions-api.rst

```diff
@@ -592,6 +685,83 @@ These base classes should be implemented by respective classes:
     class ReadOnlyPool(ReadOnlyExecutor): ...
 
 
+Subtransactions API
+'''''''''''''''''''
+
+Subtransaction API is only accessible on the transaction and subtransaction
+objects:
+
+.. code-block:: python
+
+    class AsyncIOTransaction:
+        async def subtransaction() -> AsyncIOSubtransaction: ...
+
+    class AsyncIOSubtransaction(Executor):
+        async def subtransaction() -> AsyncIOSubtransaction: ...
+        async def rollback() -> None: ...
+
+Subtransaction objects also implement ``Executor`` interface.
+
+Example using ``subtransaction``:
+
+.. code-block:: python
+
+    async for tx in pool.retry():
+        async with tx:
+            try:
+                async with tx.subtransaction() as subtx:
+                    await subtx.query('''
+                        INSERT Note { name := <str>$name, text := <str>$text }
+                    ''', name=name, text=text)
+            except ConstraintViolationError:
+                async with tx.subtransaction() as subtx:
+                    await subtx.query('''
+                        UPDATE Note
+                        FILTER .name = <str>$name
+                        SET { text := <str>$text }
+                    ''', name=name, text=text)
+
+Note:
+
+1. Transaction object is "borrowed" for the duration of subtransaction,
+   you can't execute queries on original transaction object when subtransaction
+   is active.
```

Comment:
Can we duplicate this block for the Connection API, clarifying that in

```
    async for tx in conn.retry():
        async with tx:
            try:
                async with tx.subtransaction() as subtx:
```

both `conn` qand `tx` cannot be used while `async with tx.subtransaction()` is running?

Reply:
Added. Feel free to do post-merge edits if that is not clear enough.